### DISCUSSION TITLE
Follow on to the request to get caller IP..  Also need the server name

### DISCUSSION THREAD

**Discussion by golfingal72** [apollographql/router#1262] [Q&A] [answered]
Similar to http servlet request, it has the calling server name, need that also...

So i have external authorization that might generate Http Cookies that i want to embed as a 'Set-Cookie' on the response..  In order to set the proper domain on the cookie, I need to have the incoming server name from the client request....

This would be a follow on to get the request connection information when i asked for the IP address only

Thanks

**Comment by garypen** [comment, marked as answer]
It's possible to get the HOST header from an `originating_request` member. You can also retrieve the host member from the originating_request URI details. e.g.:

```
self.originating_request.headers().get("HOST") 
self.originating_request.uri().host()
```

This will report the data that the client has provided (which may be an IP address or a DNS resolvable name or ...). There is no service within the router to perform a DNS lookup, so if the host details weren't in the form you required, you would have to provide that capability to your plugin.

I can't recommend a rust DNS lib, but there seem to be a selection to choose from.

**Reply by golfingal72** [reply]
Thanks much, that worked!

**Reply by calvinl** [reply]
@garypen forgive me for jumping into this thread, but could you provide a bit of context on how this might be used in a Rhai script to forward the caller's IP to subgraphs?

**Reply by garypen** [reply]
@calvinl Something like this snippet:

```
request.subgraph.headers["x-my-new-header"] = request.headers["HOST"];
```
(assuming you have mapped the subgraph request as per our various examples). Is that enough detail?

**Reply by calvinl** [reply]
@garypen I think I may have misunderstood this thread. I was actually looking for a way to get the client IP address. Normally, when a subgraph is queried by a user directly, the subgraph has access to the client's IP. Since we're using a federated supergraph and requests are piped through apollo router, the subgraphs lose access to the requester's IP. 

So, I don't think the HOST header is what I'm looking for. Apologies for the confusion. 

Do you know if there's a way to forward the client IP to the subgraphs? Maybe in an `x-forwarded-for` header?
